packages/next-auth/src/nextAuthPlugin.ts

```diff
@@ -62,7 +61,7 @@ export const makeNextAuthPlugin =
 
       // If the endpoint requires authentication, but no user is logged in,
       // throw unauthorized
-      if (endpoint?.config?.authenticated) {
+      if (context.endpoint?.config?.authenticated) {
```

Comment:
nit, is `endpoint` really optional here?
```suggestion
      if (context.endpoint.config?.authenticated) {
```

Reply:
Nope, good point, pushed a fix